mne/io/tests/test_meas_info.py

```diff
@@ -408,52 +408,89 @@ def test_check_consistency():
     assert_array_equal(info3['ch_names'], ['a', 'b-0', 'b-1', 'c', 'b-2'])
 
 
+def _is_anonymous(inst):
+    """Check all the anonymity fields.
+
+    inst is either a raw or epochs object.
+    """
+    from collections import namedtuple
+    anonymity_checks = namedtuple("anonymity_checks",
+                                  ["missing_subject_info",
+                                   "anonymous_file_id_secs",
+                                   "anonymous_file_id_usecs",
+                                   "anonymous_meas_id_secs",
+                                   "anonymous_meas_id_usecs",
+                                   "anonymous_meas_date",
+                                   "anonymous_annotations"])
+
+    if 'subject_info' not in inst.info.keys():
+        missing_subject_info = True
+    else:
+        missing_subject_info = inst.info['subject_info'] is None
+
+    anonymous_file_id_secs = inst.info['file_id']['secs'] == DATE_NONE[0]
+    anonymous_file_id_usecs = inst.info['file_id']['usecs'] == DATE_NONE[1]
+    anonymous_meas_id_secs = inst.info['meas_id']['secs'] == DATE_NONE[0]
+    anonymous_meas_id_usecs = inst.info['meas_id']['usecs'] == DATE_NONE[1]
+    if inst.info['meas_date'] is None:
+        anonymous_meas_date = True
+    else:
+        assert isinstance(inst.info['meas_date'], tuple)
+        anonymous_meas_date = inst.info['meas_date'] == DATE_NONE
+    if not hasattr(inst, 'annotations'):
+        anonymous_annotations = True
+    else:
+        if inst.annotations is None:
+            anonymous_annotations = True
+        else:
+            anonymous_annotations = inst.annotations.orig_time is None
```

Comment:
this block is just

```
anonymous_annotations = False
if hasattr(inst, 'annotations') and inst.annotations.orig_time is None:
    anonymous_annotations = True 
```

Reply:
For a generic `is_anonymous` you both flipped the condition of returning True when the item has no annotations, plus it breaks until #5416 gets in:

```sh
>>> class Item(object):
...     annotations = None
... 
>>> inst = Item()
>>> hasattr(inst, 'annotations') and inst.annotations.orig_time is None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'orig_time'
```

For the case that `_is_anonymous` is only a conveninet function for test_anonymize then I agree.
I'm -0 for this one. If you really want to I'll change it though, I've no problem with that.

Reply:
Just put a not if you want. It’s just much simpler to read than the double branching you have currently